pkg/prowloader/bigqueryjobs.go

```diff
@@ -0,0 +1,109 @@
+package prowloader
+
+import (
+	"context"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"google.golang.org/api/iterator"
+
+	"github.com/openshift/sippy/pkg/apis/prow"
+)
+
+func (pl *ProwLoader) fetchProwJobsFromOpenShiftBigQuery() ([]prow.ProwJob, []error) {
+
+	errs := []error{}
+
+	// Figure out our last imported job timestamp:
+	var lastProwJobRun time.Time
+	row := pl.dbc.DB.Table("prow_job_runs").Select("max(timestamp)").Row()
+	err := row.Scan(&lastProwJobRun)
+	if err != nil || lastProwJobRun.IsZero() {
+		log.WithError(err).Warn("no last prow job run found (new database?), importing last two weeks")
+		lastProwJobRun = time.Now().Add(-14 * 24 * time.Hour)
+	} else {
+		// adjust the last job run time slightly, we're querying all jobs that have completed since our last recorded
+		// job START time, but we'll subtract another 10 minutes just in case there were upload delays to gcs/bigquery:
+		lastProwJobRun = lastProwJobRun.Add(-10 * time.Minute)
+	}
```

Comment:
I don't think this is enough.  Timestamp is the job start time, so imagine this:

- Job A starts at 10am
- Job B starts at 10:15 am
- Job B completes in 15 minutes, maybe due to an infra error
- Fetchdata runs around 10:30, picks up job B

You'll never get job A from big query using this logic.

Reply:
I think we're up to allowing 8h for aggregated jobs, that's probably the minimum lookback.  To be safe I'd say even 12h.